cli/cmd/aeon.go

```diff
@@ -80,7 +84,89 @@ func NewAeonCmd() *cobra.Command {
 }
 
 func aeonConnectValidateArgs(cmd *cobra.Command, args []string) error {
-	connectCtx.Network, connectCtx.Address = libconnect.ParseBaseURI(args[0])
+
+	if len(args) == 1 && util.IsURL(args[0]) {
+		connectCtx.Network, connectCtx.Address = libconnect.ParseBaseURI(args[0])
+	} else if len(args) == 2 && util.IsRegularFile(args[0]) {
+		var (
+			configPath = args[0]
+			instance   = args[1]
+		)
+
+		f, err := os.ReadFile(configPath)
+		if err != nil {
+			return err
+		}
+
+		pb := cluster.NewYamlCollector(f)
+		config, err := pb.Collect()
+		if err != nil {
+			return err
+		}
+
+		clusterConfig, err := cluster.MakeClusterConfig(config)
+		if err != nil {
+			return err
+		}
+
+		result := cluster.Instantiate(clusterConfig, instance)
+
+		// Get SSL connection.
+		dataSsl := []string{"roles_cfg", "aeon.grpc", "advertise"}
+		data, err := result.Get(dataSsl)
+		if err != nil {
+			return err
+		}
+
+		advertise, ok := data.(map[any]any)
+		if !ok {
+			return errors.New("data it is impossible to result in a map")
+		}
+
+		uri, ok := advertise["uri"]
+		if !ok {
+			return errors.New("filed uri not found")
+		}
+
+		re := regexp.MustCompile("^https?://")
+		cleanedURL := re.ReplaceAllString(uri.(string), "")
+
+		connectCtx.Network, connectCtx.Address = libconnect.ParseBaseURI(cleanedURL)
+
+		params, ok := advertise["params"]
+		if !ok {
+			return errors.New("filed params not found")
+		}
+
+		paramsMap, ok := params.(map[any]any)
+		if !ok {
+			return errors.New("params it is impossible to result in a map")
+		}
+
+		if transport, ok := paramsMap["transport"]; ok {
+			if (transport.(string) != "ssl") && (transport.(string) != "plain") {
+				return errors.New("transport must be ssl or plain")
+			}
+
+			if transport.(string) == "ssl" {
+				connectCtx.Transport = aeoncmd.TransportSsl
+			} else {
+				connectCtx.Transport = aeoncmd.TransportPlain
+			}
+		}
+
+		if keyFile, ok := paramsMap["ssl_key_file"]; ok {
+			connectCtx.Ssl.KeyFile = keyFile.(string)
+		}
+
+		if certFile, ok := paramsMap["ssl_cert_file"]; ok {
+			connectCtx.Ssl.CertFile = certFile.(string)
+		}
+
+		if caFile, ok := paramsMap["ssl_ca_file"]; ok {
+			connectCtx.Ssl.CaFile = caFile.(string)
+		}
```

Comment:
It's better to use `mapstructure` for such boring parsing. See:

https://github.com/tarantool/tt/blob/7e746c43e67c3b59fbc5df74f3f6d878a4949adc/cli/status/status.go#L43-L70

https://github.com/tarantool/tt/blob/master/cli/status/status.go#L194

Reply:
Done